src/AstToCFlat.ml

```diff
@@ -796,10 +802,10 @@ and mk_expr (env: env) (locals: locals) (e: expr): locals * CF.expr =
       locals, CF.BufSub (e1, mk_mul32 e2 (mk_uint32 mult), base_size)
 
   | EBufDiff (e1, e2) ->
-      let mult, base_size = cell_size env (assert_buf e.typ) in
+      let mult, _ = cell_size env (assert_buf e.typ) in
       let locals, e1 = mk_expr env locals e1 in
       let locals, e2 = mk_expr env locals e2 in
-      locals, CF.BufDiff (e1, mk_mul32 e2 (mk_uint32 mult), base_size)
+      locals, mk_div32 (mk_sub32 e1 e2) (mk_uint32 mult)
```

Comment:
is it `mult` or `base_size` you need? can't remember which is which

Reply:
I think you need to compute `(e1 - e2) / (mult * bytes_in base_size)`... if I remember well, `base_size` is the type or array cells (bytes, A64s, etc.) and `mult` is how many of these base cells a single element occupies... since `e1-e2` gives you a size in bytes, you need the more involved computation